dev/io.openliberty.microprofile.telemetry.2.0.internal/src/io/openliberty/microprofile/telemetry20/internal/http/MPTelemetryHTTPMetricsAdapterImpl.java

```diff
@@ -53,7 +55,11 @@ public class MPTelemetryHTTPMetricsAdapterImpl implements HTTPMetricAdapter {
     private static final Double[] BUCKET_BOUNDARIES = { 0.005, 0.01, 0.025, 0.05, 0.075, 0.1, 0.25, 0.5, 0.75, 1.0, 2.5, 5.0, 7.5, 10.0 };
     private static final List<Double> BUCKET_BOUNDARIES_LIST = Arrays.asList(BUCKET_BOUNDARIES);
 
-    private final WeakHashMap<OpenTelemetry, DoubleHistogram> httpHistogramMap = new WeakHashMap<OpenTelemetry, DoubleHistogram>();
+    //For performance optimisation we do not use a concurrent hash map even though this class
+    //is accessed from many threads at once. So all use of threadUnsafeHTTPHistogramMap must be
+    //manually synchronised.
```

Comment:
```suggestion
    //All access to threadUnsafeHTTPHistogramMap must be syncrhonized using httpHistogramMapLock
```
The reasoning in the comment is inaccurate and we don't need that much detail anyway.

Reply:
I committed the change.